archaius2-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java

```diff
@@ -150,169 +112,143 @@ public ArchaiusModule withCascadeStrategy(Class<? extends CascadeStrategy> casca
         return this;
     }
     
-    private String getUniqueName(String prefix) {
-        uniqueNameCounter++;
-        return prefix +"-" + uniqueNameCounter;
+    protected void configureArchaius() {
```

Comment:
This follows the same approach as ServletModule where sub-classing the module is only done for customization purposes.  Note that InternalArchaiusModule is final and has the proper deduping equals/hashCode.